sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs

```diff
@@ -33,58 +33,60 @@ public static class AnalysisContextExtensions
 
         public static void ReportDiagnosticWhenActive(this SyntaxNodeAnalysisContext context, Diagnostic diagnostic)
         {
-            ReportWhenNotSuppressed(context.Node.SyntaxTree, diagnostic, d => context.ReportDiagnostic(d));
+            TryReportDiagnostic(context.Node.SyntaxTree, diagnostic, d => context.ReportDiagnostic(d));
         }
 
         public static void ReportDiagnosticWhenActive(this SyntaxTreeAnalysisContext context, Diagnostic diagnostic)
         {
-            ReportWhenNotSuppressed(context.Tree, diagnostic, d => context.ReportDiagnostic(d));
+            TryReportDiagnostic(context.Tree, diagnostic, d => context.ReportDiagnostic(d));
         }
 
         public static void ReportDiagnosticWhenActive(this CompilationAnalysisContext context, Diagnostic diagnostic)
         {
-            ReportWhenNotSuppressed(context.Compilation.SyntaxTrees.FirstOrDefault(), diagnostic,
+            TryReportDiagnostic(context.Compilation.SyntaxTrees.FirstOrDefault(), diagnostic,
                 d => context.ReportDiagnostic(d));
         }
 
         public static void ReportDiagnosticWhenActive(this SymbolAnalysisContext context, Diagnostic diagnostic)
         {
-            ReportWhenNotSuppressed(context.Symbol.Locations.FirstOrDefault(l => l.SourceTree != null)?.SourceTree, diagnostic,
+            TryReportDiagnostic(context.Symbol.Locations.FirstOrDefault(l => l.SourceTree != null)?.SourceTree, diagnostic,
                 d => context.ReportDiagnostic(d));
         }
 
         public static void ReportDiagnosticWhenActive(this CodeBlockAnalysisContext context, Diagnostic diagnostic)
         {
-            ReportWhenNotSuppressed(context.CodeBlock.SyntaxTree, diagnostic, d => context.ReportDiagnostic(d));
+            TryReportDiagnostic(context.CodeBlock.SyntaxTree, diagnostic, d => context.ReportDiagnostic(d));
         }
 
-        private static void ReportWhenNotSuppressed(SyntaxTree tree, Diagnostic diagnostic, Action<Diagnostic> report)
+        private static void TryReportDiagnostic(SyntaxTree tree, Diagnostic diagnostic, Action<Diagnostic> report)

```

Comment:
Why "Try" in the name? It doesn't follow the usual "try" pattern. Perhaps simply `ReportDiagnostics` or `ReportDiagnosticsWhenEnabled`?

Reply:
Because I couldn't find any better name. `ReportDiagnosticsWhenEnabled` doesn't handle the suppression :c

Reply:
Or something like _InternalReportDiagnostic_

Reply:
So maybe just `ReportDiagnostics` then?

Reply:
`Diagnostics` suggest a collection while we are providing only one. I will go with `InternalReportDiagnostic` as I expect this class to be changed soon anyway.